pkg/ast/spl/spl.peg

```diff
@@ -951,6 +956,301 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
+        return nil, fmt.Errorf("Invalid limit (%v): %v", intAsStr.(string), err)
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "bins",
+        binOptions: &structs.BinCmdOptions {
+            MaxBins: numBins,
+        },
+    }
+    return binOptionArgs, nil
+}
+
+PositiveIntegerAsString <- [0-9]+ {
+    return string(c.text), nil
+}
+
+// TODO: chain modifier and what else is supported?
+
+BinOptionAlignTime <- "aligntime" EQUAL utcEpoch:(PositiveIntegerAsString) {
+    epoch, err := strconv.ParseUint(utcEpoch.(string), 10, 64)
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Invalid utc epoch value: %v for align time", utcEpoch.(string))
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "aligntime",
+        binOptions: &structs.BinCmdOptions {
+            AlignTime: &epoch,
+        },
+    }
+
+    return binOptionArgs, nil
+}
+/ "aligntime" EQUAL timestamp:(AbsoluteTimestamp / RelativeTimestamp) {
+    var epoch uint64 = 0
+    var err error
+    relTimeModifier := timestamp.(ast.TimeModifier)
+    epochVal, err := CalculateRelativeTime(relTimeModifier, time.Now())
+    epoch = uint64(epochVal)
+    if err != nil {
+        return nil, err
+    }
+
+    binOptionArgs := &BinOptionArgs {
+        argOption: "aligntime",
+        binOptions: &structs.BinCmdOptions {
+            AlignTime: &epoch,
+        },
+    }
+    
+    return binOptionArgs, nil
+}
+
+BinOptionStart <- "start" EQUAL number:(FloatAsString / IntegerAsString) {
+    start, err := strconv.ParseFloat(number.(string), 64)
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Cannot convert %v to float", number.(string))
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "start",
+        binOptions: &structs.BinCmdOptions {
+            Start: &start,
+        },
+    }
+
+    return binOptionArgs, nil
+}
+
+BinOptionEnd <- "end" EQUAL number:(FloatAsString / IntegerAsString) {
+    end, err := strconv.ParseFloat(number.(string), 64)
+    if err != nil {
+        return nil, fmt.Errorf("Spl peg: Cannot convert %v to float", number.(string))
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "end",
+        binOptions: &structs.BinCmdOptions {
+            End: &end,
+        },
+    }
+
+    return binOptionArgs, nil
+}
+
+BinOptionSpan <- "span" EQUAL num1:(FloatAsString / IntegerAsString)? "log" num2:(FloatAsString / IntegerAsString)? {
+    var coeff float64 = 1.0
+    var base float64 = 10.0
+    var err error
+
+    if num2 != nil {
+        base, err = strconv.ParseFloat(num2.(string), 64)
+        if err != nil || base <= 1.0 {
+            return nil, fmt.Errorf("Spl peg: Invalid base in log span, err: %v", err)
+        }
+    }
+    
+    if num1 != nil {
+        coeff, err = strconv.ParseFloat(num1.(string), 64)
+        if err != nil || coeff < 1.0 || coeff >= base {
+            return nil, fmt.Errorf("Spl peg: Invalid coefficient in log span, err: %v", err)
```

Comment:
Split up these error cases so if one of them happens, it's more clear what the actual error is. For example, on 1104 if `base <= 1.0` the error we'd get is
```
Spl peg: Invalid base in log span, err: <nil>
```
which isn't very clear

Reply:
addressed in the latest commit.